lib/internal/modules/run_main.js

```diff
@@ -68,22 +70,27 @@ function shouldUseESMLoader(mainPath) {
   if (mainPath && StringPrototypeEndsWith(mainPath, '.mjs')) { return true; }
   if (!mainPath || StringPrototypeEndsWith(mainPath, '.cjs')) { return false; }
 
-  const { readPackageScope } = require('internal/modules/package_json_reader');
-  const pkg = readPackageScope(mainPath);
-  // No need to guard `pkg` as it can only be an object or `false`.
-  switch (pkg.data?.type) {
-    case 'module':
-      return true;
-    case 'commonjs':
-      return false;
-    default: { // No package.json or no `type` field.
-      if (getOptionValue('--experimental-detect-module')) {
-        // If the first argument of `containsModuleSyntax` is undefined, it will read `mainPath` from the file system.
-        return containsModuleSyntax(undefined, mainPath);
-      }
-      return false;
+  const response = getNearestParentPackageJSONType(path.toNamespacedPath(mainPath));
```

Comment:
const response = getNearestParentPackageJSONType(mainPath);
```suggestion
  const response = getNearestParentPackageJSONType(path.toNamespacedPath(mainPath));
```

Reply:
With this change, I am getting that ` .\out\Release\node.exe  ./test/pummel/test-policy-integrity-parent-module.js` succeeds whereas it fails in the current PR.

Reply:
Thanks, I've forced pushed with this change. Will re-run the CI.